templates/scripts/man/terminal.man
.\" Manpage for GEN_SCRIPT_REPLACE_NEWFILE_SHORT
.TH man 1 "14 July 2022" "GEN_SCRIPT_REPLACE_VERSION" "GEN_SCRIPT_REPLACE_NEWFILE_SHORT"

.SH NAME
GEN_SCRIPT_REPLACE_NEWFILE_SHORT

.SH DESCRIPTION
GEN_SCRIPT_REPLACE_DESC

.SH SYNOPSIS
GEN_SCRIPT_REPLACE_NEWFILE_SHORT [options] [command]

.SH OPTIONS
.TP
.B \-\-help
Display the full help.
.TP
.B \-\-version
Display current version.
.TP
.B \-\-config
Create user settings file.
.TP
.B \-\-options
Show all availabe options.
.TP
.B \-\-dir
Change current directory
.TP
.B \-\-debug
Output debugging information.
.TP
.B \-\-raw
Removes all formatting

.SH EXAMPLES
.TP
.B GEN_SCRIPT_REPLACE_NEWFILE_SHORT \-\-help

.SH FILES
.TP
.I
~/.config/myscripts/GEN_SCRIPT_REPLACE_NEWFILE_SHORT/settings.conf
.TP
.I
/usr/local/bin/GEN_SCRIPT_REPLACE_NEWFILE_SHORT

.SH TODO
GEN_SCRIPT_REPLACE_TODO

.SH LICENSE
GEN_SCRIPT_REPLACE_LICENSE

.SH BUGS
No known bugs.

.SH REPORTING BUGS
https://github.com/casjay-dotfiles/issues

.SH AUTHOR
Currently maintained by GEN_SCRIPT_REPLACE_AUTHOR <GEN_SCRIPT_REPLACE_EMAIL>


CHANGELOG.md
#### Changelog: 2022-08-01 at 19:54:24  
#### 🗃️ Committing everything that changed 🗃️  
  
functions/global/menu.bash      🚀  
functions/global/processes.bash      🚀  
  
#### End of changes for 202208011515-git - 2022-08-01 at 19:54:24  
  
----  
  
#### Changelog: 2022-08-01 at 15:15:12  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  
version.txt      🚀  
  
#### End of changes for  - 2022-08-01 at 15:15:12  
  
----  
  
#### Changelog: 2022-08-01 at 15:14:01  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  
  
#### End of changes for  - 2022-08-01 at 15:14:01  
  
----  
  
#### Changelog: 2022-07-30 at 12:58:09  
#### 🗃️ Committing everything that changed 🗃️  
  
bin/pkmgr      🚀  
  
#### End of changes for  - 2022-07-30 at 12:58:09  
  
----  
  
#### Changelog: 2022-07-29 at 11:36:11  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  
  
#### End of changes for 202207290819-git - 2022-07-29 at 11:36:11  
  
----  
  
#### Changelog: 2022-07-29 at 08:19:08  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  
templates/scripts/other/build      🚀  
templates/scripts/shell/bash      🚀  
  
#### End of changes for 202207272202-git - 2022-07-29 at 08:19:08  
  
----  
  
#### Changelog: 2022-07-27 at 22:02:28  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  
  
#### End of changes for 202207271327-git - 2022-07-27 at 22:02:28  
  
----  
  
#### Changelog: 2022-07-27 at 13:27:15  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  
  
#### End of changes for 202207271212-git - 2022-07-27 at 13:27:15  
  
----  
  
#### Changelog: 2022-07-27 at 12:12:13  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
  
#### End of changes for 202207271206-git - 2022-07-27 at 12:12:13  
  
----  
  
#### Changelog: 2022-07-27 at 12:06:26  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
  
#### End of changes for 202207271201-git - 2022-07-27 at 12:06:26  
  
----  
  
#### Changelog: 2022-07-27 at 12:03:35  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
  
#### End of changes for 202207271201-git - 2022-07-27 at 12:03:35  
  
----  
  
#### Changelog: 2022-07-27 at 12:01:24  
#### 🗃️ Committing everything that changed 🗃️  
  
functions/app-installer.bash      🚀  
functions/mgr-installers.bash      🚀  
  
#### End of changes for 202207271152-git - 2022-07-27 at 12:01:24  
  
----  
  
#### Changelog: 2022-07-27 at 11:52:22  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
  
#### End of changes for 202207271053-git - 2022-07-27 at 11:52:22  
  
----  
  
#### Changelog: 2022-07-27 at 10:53:02  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  
  
#### End of changes for 202207271038-git - 2022-07-27 at 10:53:02  
  
----  
  
#### Changelog: 2022-07-27 at 10:38:54  
#### 🗃️ Committing everything that changed 🗃️  
  
functions/types/main.bash      🚀  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  
  
#### End of changes for 202207270909-git - 2022-07-27 at 10:38:54  
  
----  
  
#### Changelog: 2022-07-27 at 09:09:53  
#### 🗃️ Committing everything that changed 🗃️  
  
functions/app-installer.bash      🚀  
functions/types/main.bash      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
  
#### End of changes for 202207270322-git - 2022-07-27 at 09:09:53  
  
----  
  
#### Changelog: 2022-07-27 at 03:22:54  
#### 🗃️ Committing everything that changed 🗃️  
  
templates/scripts/bash/mgr-script.system.sh      🚀  
templates/scripts/bash/mgr-script.user.sh      🚀  
templates/scripts/bash/simple      🚀  
templates/scripts/bash/system      🚀  
templates/scripts/bash/terminal      🚀  
templates/scripts/bash/user      🚀  

templates/scripts/completions/terminal.bash
#!/usr/bin/env bash
# shellcheck shell=bash
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
##@Version           :  GEN_SCRIPT_REPLACE_VERSION
# @@Author           :  GEN_SCRIPT_REPLACE_AUTHOR
# @@Contact          :  GEN_SCRIPT_REPLACE_EMAIL
# @@License          :  GEN_SCRIPT_REPLACE_LICENSE
# @@ReadME           :  GEN_SCRIPT_REPLACE_FILENAME --help
# @@Copyright        :  GEN_SCRIPT_REPLACE_COPYRIGHT
# @@Created          :  GEN_SCRIPT_REPLACE_DATE
# @@File             :  GEN_SCRIPT_REPLACE_FILENAME
# @@Description      :  GEN_SCRIPT_REPLACE_DESC
# @@Changelog        :  GEN_SCRIPT_REPLACE_CHANGELOG
# @@TODO             :  GEN_SCRIPT_REPLACE_TODO
# @@Other            :  GEN_SCRIPT_REPLACE_OTHER
# @@Resource         :  GEN_SCRIPT_REPLACE_RES
# @@Terminal App     :  GEN_SCRIPT_REPLACE_TERMINAL
# @@sudo/root        :  GEN_SCRIPT_REPLACE_SUDO
# @@Template         :  completions/system
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
_GEN_SCRIPT_REPLACE_FILENAME_completion() {
  ___findcmd() { find -L "${1:-$CONFDIR/}" -maxdepth ${3:-3} -type ${2:-f} 2>/dev/null | sed 's#'${1:-$CONFDIR}'##g' | grep '^' || return 1; }
  local cur prev words cword opts split
  local cur="${COMP_WORDS[$COMP_CWORD]}"
  local prev="${COMP_WORDS[$COMP_CWORD - 1]}"
  local CONFFILE="settings.conf"
  #####################################################################
  local CONFDIR="$HOME/.config/myscripts/GEN_SCRIPT_REPLACE_FILENAME"
  local SEARCHDIR="${CONFDIR:-$HOME/.config/myscripts/GEN_SCRIPT_REPLACE_FILENAME}"
  #####################################################################
  local SHOW_COMP_OPTS=""
  #####################################################################
  local SHORTOPTS=""
  #####################################################################
  local LONGOPTS="--completions --config --debug --dir --help --options --raw --version --term "
  #####################################################################
  local ARRAY=""
  #####################################################################
  local LIST=""
  #####################################################################
  _init_completion || return
  #####################################################################
  if [ "$SHOW_COMP_OPTS" != "" ]; then
    local SHOW_COMP_OPTS_SEP="$(echo "$SHOW_COMP_OPTS" | tr ',' ' ')"
    compopt -o $SHOW_COMP_OPTS_SEP
  fi
  #####################################################################
  if [[ ${cur} == --* ]]; then
    COMPREPLY=($(compgen -W '${LONGOPTS}' -- ${cur}))
  elif [[ ${cur} == -* ]]; then
    COMPREPLY=($(compgen -W '${SHORTOPTS:---}' -- ${cur})) && compopt -o nospace
  else
    case "${prev:-${COMP_WORDS[1]}}" in
    --completions)
      prev=""
      COMPREPLY=($(compgen -W 'long short list array' -- "$cur"))
      ;;
    --debug | --raw | --help | --version | --config | --options)
      COMPREPLY=($(compgen -W '${ARRAY} ${LONGOPTS} ${SHORTOPTS}' -- ${cur}))
      return 0
      ;;
    --dir)
      prev="dir"
      [ "$cword" -le 2 ] && _filedir -d ||
        COMPREPLY=($(compgen -W '${ARRAY}' -- "$cur"))
      ;;
    *)
      COMPREPLY=($(compgen -W '${ARRAY}' -- "$cur"))
      return 0
      ;;
    esac
  fi
  #
  # if [ -n "$FILEDIR" ]; then _filedir; fi
  # if [ $cword -gt 2 ]; then
  #   return
  # elif [ $cword == 2 ]; then
  #   _filedir
  #   compopt -o nospace
  #   return
  # elif [ $cword -eq 1 ]; then
  #   COMPREPLY=($(compgen -W '{a..z}{a..z}' -- "${cur}"))
  #   compopt -o nospace
  #   return
  # if [ "$ARRAY" = "show__none" ]; then
  #   COMPREPLY=($(compgen -W '' -- "${cur}"))
  # elif [ "$ARRAY" = "show__filedir" ]; then
  #   _filedir
  # elif [ "$ARRAY" = "show__commands" ]; then
  #   COMPREPLY=($(compgen -c -- "${cur}"))
  # elif [ "$ARRAY" != "" ]; then
  #   COMPREPLY=($(compgen -W '${ARRAY}' -- "${cur}"))
  # elif [ -n "$OPTS" ]; then
  #   COMPREPLY=($(compgen -W '${OPTS}' -- "${cur}"))
  # else
  #   COMPREPLY=($(compgen -W '${ARRAY}' -- "${cur}"))
  # elif [ $cword -gt 2 ]; then
  #   return
  # elif [ $cword == 2 ]; then
  #   _filedir
  #   compopt -o nospace
  #   return
  # elif [ $cword -eq 1 ]; then
  #   COMPREPLY=($(compgen -W '{a..z}{a..z}' -- "${cur}"))
  #   compopt -o nospace
  #   return
  # fi
  # [ ${cword} == 2 ] && _filedir && compopt -o nospace
  # [ $COMP_CWORD -eq 2 ] && COMPREPLY=($(compgen -W '{a..z} {A..Z} {0..9}' -o nospace -- "${cur}"))
  # [ $COMP_CWORD -eq 3 ] && COMPREPLY=($(compgen -W '$(_filedir)' -o filenames -o dirnames -- "${cur}"))
  # [ $COMP_CWORD -gt 3 ] && COMPREPLY=($(compgen -W '' -- "${cur}"))
  # prev=""
  # compopt -o nospace
} &&
  # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
  # enable completions
  complete -F _GEN_SCRIPT_REPLACE_FILENAME_completion GEN_SCRIPT_REPLACE_FILENAME
